### PR TITLE
Adding Kevlar Shears

### PR DESCRIPTION
**Pull request by a-chancey**
This adds kevlar shears to the game, a new Cutting Quality 2 tool, as well as putting them in appropriate spawn groups and giving them a crafting recipe.
This also updates the tailoring requirements for kevlar to include cutting 2, which is the shears.

#### Summary
Bugfixes "Adding Kevlar shears"

#### Purpose of change

This is the first PR towards #53315, adding in Kevlar Shears. This doesn't fix the reverse recipe issues that sewing 2+ requires cut 2+ to reverse, but that is beyond the scope of this PR.

#### Describe the solution

This gives us a cut quality 2 tool that will cover cutting Kevlar, as it requires specialized shears in real life to cut.
The Kevlar shears were given a recipe similar to scissors, but with a higher difficulty and time required, as well as a higher penalty for not having the bladesmithing proficiency, since they're a different type of blade *and* serrated.
The PR also increases tailoring requirements for Kevlar working.

#### Describe alternatives you've considered

reducing the recipes to use Cut 1, adding alternative Cut 2 items (soon), adjusting the reverse recipe function to pull from a defined list of valid qualities.

#### Testing

Spawned in shears and Kevlar gear, was able to disassemble it just fine. Teleported to a few loot spawn locations (mansions, sewing craft store) and found the shears there.

#### Additional context

There's still an issue with reversing recipes that a quality can appear twice, I'll file an issue on this later.

### CODE REVIEW

File: data/json/itemgroups/Locations_MapExtras/mansion.json [hunk 1]
```diff
@@ -1302,7 +1302,8 @@
       [ "rag", 30 ],
       [ "lycra_patch", 15 ],
       [ "nylon", 10 ],
-      { "item": "tailors_kit", "prob": 20, "charges": [ 0, 100 ] }
+      { "item": "tailors_kit", "prob": 20, "charges": [ 0, 100 ] },
+      [ "kevlar_shears", 30 ]
```
Comment (I-am-Erk) [inline data/json/itemgroups/Locations_MapExtras/mansion.json:1306]:
```suggestion
      [ "kevlar_shears", 5 ]
```
I think the chance of finding kevlar shears in a mansion is significantly lower than finding a tailor's kit.

File: data/json/itemgroups/SUS/clothes_store.json [hunk 2]
```diff
@@ -69,7 +69,8 @@
               { "item": "needle_curved", "prob": 75 },
               { "item": "scissors", "prob": 90 },
               { "item": "knitting_needles", "prob": 20 },
-              { "item": "sewing_kit", "prob": 85, "charges-min": 0 }
+              { "item": "sewing_kit", "prob": 85, "charges-min": 0 },
+              { "item": "kevlar_shears", "prob": 75 }
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/clothes_store.json:73]:
```suggestion
              { "item": "kevlar_shears", "prob": 5 }
```
Again, this is a place you might find them, but not on anything resembling the frequency you'd find a sewing kit. Even this is exceptionally high.

File: data/json/itemgroups/SUS/domestic.json [hunk 3]
```diff
@@ -306,6 +306,7 @@
       { "item": "stapler", "prob": 25 },
       { "item": "soldering_iron", "prob": 25, "charges": [ 5, 45 ] },
       { "item": "xacto", "prob": 25 },
+      { "item": "kevlar_shears", "prob": 25 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/SUS/domestic.json:309]:
```suggestion
      { "item": "kevlar_shears", "prob": 10 },
```

File: data/json/itemgroups/tools.json [hunk 4]
```diff
@@ -547,7 +548,8 @@
       [ "needle_curved", 30 ],
       [ "scissors", 1000 ],
       { "item": "sewing_kit", "prob": 10, "charges-min": 0 },
-      { "item": "tailors_kit", "prob": 120, "charges-min": 0 }
+      { "item": "tailors_kit", "prob": 120, "charges-min": 0 },
+      [ "kevlar_shears", 500 ]
```
Comment (I-am-Erk) [inline data/json/itemgroups/tools.json:552]:
```suggestion
      [ "kevlar_shears", 40 ]
```
Again, compare to what I'd call the flagship item of the set, a tailor's kit should be more common a spawn than kevlar shears in this group.

File: data/json/items/tool/tailoring.json [hunk 5]
```diff
@@ -255,6 +255,25 @@
     "symbol": ";",
     "color": "yellow"
   },
+  {
+    "id": "kevlar_shears",
+    "type": "TOOL",
+    "name": { "str": "pair of Kevlar shears", "str_pl": "pairs of Kevlar shears" },
+    "description": "These are a pair of heavy-duty shears with a blade suitable for cutting Kevlar.  The blades feature a flat shearing edge and a fine serration, and are excellent at cutting Kevlar and other polymer based fabrics.",
```
Comment (I-am-Erk) [inline data/json/items/tool/tailoring.json:262]:
```suggestion
    "description": "These are a pair of heavy-duty shears with a blade suitable for cutting particularly tough materials, from polyaramid fabrics to tough plastics and thin metals.  The blades feature a flat shearing edge and a fine serration.",
```

**Review by I-am-Erk — COMMENTED** [review]
Mostly looking good, but I think the frequencies are rather high here. Even my suggested frequencies may be too much.

**Comment by a-chancey** [comment]
> Mostly looking good, but I think the frequencies are rather high here. Even my suggested frequencies may be too much.

I'll tone them down and check them at your suggested rates. If they still seem to frequent I can bump them down - after all, they are craftable. 

In the issue I mentioned also reworking sewing and tailors kits slightly as well, if you have any feedback it would be appreciated.

A sewing kit is going to need some sort of cutting tool in order to trim fabric and cut thread. I was considering adding scissors or one of a handful of knives as an ingredient and giving the sewing kit CUT 1 as well. A quick search shows that pretty much any commercially available sewing kit will come with at least small scissors. 

A tailor's kit includes more specialized tools - so I was considering adding the Kevlar shears to it and giving it CUT 2. 

Based on my reading, Nomex should actually require a similar type of shear to cut smoothly as well, and it was suggested to me that it and synthetic fabric also be given the CUT 2 requirement. This would help reduce that extra inventory item later on, and I think has realistic reasoning behind the change.

**Pull request merged**